tests/results/test_result_factory.py

```diff
@@ -82,6 +83,24 @@ def foo():
     assert result_factory.persist_result is True
 
 
+def test_root_flow_default_remote_storage():
+    @flow
+    def foo():
+        return get_run_context().result_factory
+
+    fs = LocalFileSystem(basepath="memory/root")
+
+    with temporary_settings(
+        {
+            PREFECT_RESULTS_PERSIST_BY_DEFAULT: True,
+            PREFECT_DEFAULT_RESULT_STORAGE_BLOCK: fs.basepath,
```

Comment:
this is confusing to me, as `memory/root` is not a valid block-type/block-document-name ,  nor is it a valid basepath that specifies a protocol

it seems like whatever you set as `PREFECT_DEFAULT_RESULT_STORAGE_BLOCK` should be something like `s3/my-result-storage`